src/Microsoft.DotNet.Interactive.Http/Model/HttpNamedRequest.cs

```diff
@@ -233,11 +234,32 @@ internal HttpNamedRequest(HttpRequestNode httpRequestNode, HttpResponse response
     {
         if (currentIndex + 1 == path.Length)
         {
-            var result = responseJSON[path[currentIndex]];
-            return result?.ToString();
+            if(responseJSON.GetType() == typeof(JsonArray))
+            {
+                var jsonArray = (JsonArray)responseJSON;
+                return jsonArray.FirstOrDefault(n => n?[path[currentIndex]] != null)?.ToJsonString();
+            } 
+            else if(responseJSON.GetType() == typeof(JsonObject))
+            {
+                var jsonObject = (JsonObject)responseJSON;
+                return jsonObject[path[currentIndex]]?.ToString();
+            }
+            else
+            {
+                return null;
+            }
         }
 
-        var newResponseJSON = responseJSON[path[currentIndex + 1]];
+        JsonNode? newResponseJSON = null;
+        try
+        {
+            newResponseJSON = responseJSON[path[currentIndex]];
+        } 
+        catch (Exception)
```

Comment:
It's not ideal to catch `Exception` because it can easily hide bugs. If there's a specific exception type you're expecting here, it's better to catch that.